src/postprocess/dos.jl

```diff
@@ -23,8 +23,8 @@ and temperature `T`. It increases with both `T` and better sampling of the BZ wi
 Fermi surface.
 """
 function compute_nos(ε, basis, eigenvalues; smearing=basis.model.smearing,
-                     temperature=basis.model.temperature, spins=1:basis.model.n_spin_components)
-    N = zero(ε)
+                     temperature=basis.model.temperature)
+    N = zeros(typeof(ε), basis.model.n_spin_components)
```

Comment:
If you change this you certainly also have to change the way things are summed into the N

Reply:
woops... we should have a test for this.

Reply:
Yeah I did